packages/@lwc/engine-core/src/framework/update-component-value.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { VM } from './vm';
+import { componentValueMutated } from './mutation-tracker';
+
+export function updateComponentValue(vm: VM, key: string, newValue: any) {
+    const { cmpFields } = vm;
+    if (newValue !== cmpFields[key]) {
+        cmpFields[key] = newValue;
+
+        componentValueMutated(vm, key);
```

Comment:
The `componentValueMutated` method is also invoked in [decorators/api.ts](https://github.com/salesforce/lwc/blob/4b8b03e0b4246205984a020c4fb59e92a18ad4ac/packages/%40lwc/engine-core/src/framework/decorators/api.ts#L69-L71). However, always invoke `componentValueMutated` even if the value is the same. Do you know why we don't do a similar check?

Reply:
@pmdartus That's a great question. I assumed there was a good reason for it, but maybe it was just a mistake. Let me correct it and we'll see if Nucleus finds any errors.

Reply:
Ah nevermind, I just realized the difference. One is updating `vm.cmpProps`, the other is updating `vm.cmpFields`. So it's two separate routines.

That said, we could probably have the `if` equality check here.

Reply:
Hm interestingly, if we add the `if` equality check, then this test fails:

https://github.com/salesforce/lwc/blob/4b8b03e0b4246205984a020c4fb59e92a18ad4ac/packages/%40lwc/integration-karma/test/template/properties/index.spec.js#L29

So maybe we shouldn't change the logic here after all (at least in this PR).